include/SampleFrame.h

```diff
@@ -226,6 +231,18 @@ inline void copyFromSampleFrames(float* target, const SampleFrame* source, size_
 	}
 }
 
+// Enable SampleFrame to use the AudioDataType metafunction
+namespace detail {
+template<> struct AudioDataType<AudioDataKind::SampleFrame> { using type = SampleFrame; };
+} // namespace detail
+
+//! A non-owning SampleFrame buffer (interleaved, 2-channel)
+using CoreAudioData = std::span<const SampleFrame>;
+
+//! Mutable CoreAudioData
+using CoreAudioDataMut = std::span<SampleFrame>;
```

Comment:
Can we branch the `std::span` changes into another PR? It doesn't seem strictly necessary to do here, and we would have to apply it across the codebase, so that's IMO a separate effort.

I also don't like the unnecessary abstraction (`CoreAudioData` and `CoreAudioDataMut`). Using `std::span` directly is a lot clearer IMO.

Reply:
Done in https://github.com/LMMS/lmms/pull/7459/commits/f3a8b3f3f33d1b06dca907f24ae98e2ae2c0d185